sdks/python/apache_beam/runners/common.py

```diff
@@ -168,22 +169,41 @@ def __init__(self, obj_to_invoke, method_name):
     self.has_userstate_arguments = False
     self.state_args_to_replace = {}
     self.timer_args_to_replace = {}
+    self.timestamp_arg_name = None
+    self.window_arg_name = None
+
     for kw, v in zip(args[-len(defaults):], defaults):
       if isinstance(v, core.DoFn.StateParam):
         self.state_args_to_replace[kw] = v.state_spec
         self.has_userstate_arguments = True
       elif isinstance(v, core.DoFn.TimerParam):
         self.timer_args_to_replace[kw] = v.timer_spec
         self.has_userstate_arguments = True
+      elif isinstance(v, core._DoFnParam) and \
+              v.param_id == core.DoFn.TimestampParam.param_id:
+        self.timestamp_arg_name = kw
```

Comment:
Should it be an error if this is already set? (Also below.)

Reply:
I think this has already taken care.